### ISSUE TITLE
DB locks are not released

### ISSUE THREAD

**Issue by benoit74** [openzim/zimfarm#826]
Location: API

Issue https://github.com/openzim/zimfarm/issues/824 showed that we have pending DB locks for many hours.

It is not cristal clear what happened (repro to do in dev environment) but:
- the locks are all linked to "UPDATE task SET ..." statement
- the locks seems to be rooted to a situation where:
  - the API updated a task
  - the update has been flushed (not yet committed) to the DB
  - ~~an error later happened and an error occured during the exception handling, leading to a fatal issue~~
  - all subsequent API requests on the same task have timed-out
- few hours later, the API is completely stuck
  - many more locks, still all on "UPDATE task SET ..." statements
  - HTTP queue is full

What is weird is that looking at `pg_locks` view, most locks on the tasks table are ExclusiveLock but the whole table is not locked, some task achieved to start and hence insert a row and then update their status many times.

```
EXCLUSIVE (ExclusiveLock)

    Conflicts with the ROW SHARE, ROW EXCLUSIVE, SHARE UPDATE EXCLUSIVE, SHARE, SHARE ROW EXCLUSIVE, EXCLUSIVE, and ACCESS EXCLUSIVE lock modes. This mode allows only concurrent ACCESS SHARE locks, i.e., only reads from the table can proceed in parallel with a transaction holding this lock mode.

    Acquired by REFRESH MATERIALIZED VIEW CONCURRENTLY.
```

**Comment by benoit74** [comment]
This is not reproduced so far + probably wrong, no fatal error is raised in exception handling in the original issue.

**Comment by benoit74** [comment]
A probable cause would be the zmq broadcaster which has been remove in #827.
Let's wait for another case in production.

**Comment by rgaudin** [comment]
I'm skeptical about the link to broadcaster ; let's hope I'm wrong 😉

**Comment by benoit74** [comment]
I'm skeptical too, but I have faith (and no other rational explanation) 😆

**Comment by benoit74** [comment]
Looks like it is not happening anymore, so I will close this ticket.
I have opened another more generic issue regarding the fact that timeouts in general are not very properly handled: https://github.com/openzim/zimfarm/issues/832
And I have also opened another issue regarding better postgresql monitoring: https://github.com/kiwix/k8s/issues/125
